### ISSUE TITLE
spring-cloud-starter-alibaba-nacos-discovery 2.2.3.RELEASE is not compatible with Undertow

### ISSUE THREAD

**Issue by FalconIA** [alibaba/spring-cloud-alibaba#1776]
**Which Component**
Nacos Discovery

**Describe the bug**
spring-cloud-starter-alibaba-nacos-discovery 2.2.3.RELEASE is not compatible with Undertow.

**To Reproduce**
1. Stop a service using `spring-boot-starter-undertow:2.3.3.RELEASE`.
2. Got a exception when stop `nacosWatch`.
```
[ncp-example-feign-order-biz#0.0.0.0#0] [000] 2020-10-12 10:48:03,300 [SpringContextShutdownHook] O [o.s.scheduling.concurrent.ThreadPoolTaskScheduler] ExecutorConfigurationSupport.java:218 - Shutting down ExecutorService 'Nacos-Watch-Task-Scheduler'
[ncp-example-feign-order-biz#0.0.0.0#0] [000] 2020-10-12 10:48:03,305 [SpringContextShutdownHook] N [o.s.context.support.DefaultLifecycleProcessor] DefaultLifecycleProcessor.java:264 - Failed to stop bean 'nacosWatch'
java.lang.NullPointerException: null
	at io.undertow.servlet.spec.ServletContextImpl.getInitParameter(ServletContextImpl.java:425)
	at org.springframework.web.context.support.ServletContextPropertySource.getProperty(ServletContextPropertySource.java:47)
	at org.springframework.web.context.support.ServletContextPropertySource.getProperty(ServletContextPropertySource.java:33)
	at org.springframework.core.env.PropertySourcesPropertyResolver.getProperty(PropertySourcesPropertyResolver.java:85)
	at org.springframework.core.env.PropertySourcesPropertyResolver.getPropertyAsRawString(PropertySourcesPropertyResolver.java:74)
	at org.springframework.util.PropertyPlaceholderHelper.parseStringValue(PropertyPlaceholderHelper.java:151)
	at org.springframework.util.PropertyPlaceholderHelper.replacePlaceholders(PropertyPlaceholderHelper.java:124)
	at org.springframework.core.env.AbstractPropertyResolver.doResolvePlaceholders(AbstractPropertyResolver.java:239)
	at org.springframework.core.env.AbstractPropertyResolver.resolvePlaceholders(AbstractPropertyResolver.java:202)
	at org.springframework.core.env.AbstractEnvironment.resolvePlaceholders(AbstractEnvironment.java:566)
	at com.alibaba.spring.util.PropertySourcesUtils.getSubProperties(PropertySourcesUtils.java:110)
	at com.alibaba.spring.util.PropertySourcesUtils.getSubProperties(PropertySourcesUtils.java:57)
	at com.alibaba.cloud.nacos.NacosDiscoveryProperties.enrichNacosDiscoveryProperties(NacosDiscoveryProperties.java:616)
	at com.alibaba.cloud.nacos.NacosDiscoveryProperties.getNacosProperties(NacosDiscoveryProperties.java:610)
	at com.alibaba.cloud.nacos.discovery.NacosWatch.stop(NacosWatch.java:166)
	at com.alibaba.cloud.nacos.discovery.NacosWatch.stop(NacosWatch.java:98)
	at org.springframework.context.support.DefaultLifecycleProcessor.doStop(DefaultLifecycleProcessor.java:238)
	at org.springframework.context.support.DefaultLifecycleProcessor.access$300(DefaultLifecycleProcessor.java:53)
	at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.stop(DefaultLifecycleProcessor.java:377)
	at org.springframework.context.support.DefaultLifecycleProcessor.stopBeans(DefaultLifecycleProcessor.java:210)
	at org.springframework.context.support.DefaultLifecycleProcessor.onClose(DefaultLifecycleProcessor.java:128)
	at org.springframework.context.support.AbstractApplicationContext.doClose(AbstractApplicationContext.java:1022)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.doClose(ServletWebServerApplicationContext.java:170)
	at org.springframework.context.support.AbstractApplicationContext$1.run(AbstractApplicationContext.java:949)
```

**Expected behavior**
Nacos Discovery Starter compatible with Undertow.

**Additional context**
JDK: Dragonwell 8.4.4
Undertow: 2.1.3.Final
spring-cloud-dependencies: Hoxton.SR8
spring-boot: 2.3.3.RELEASE
spring: 5.2.8.RELEASE
spring-cloud-alibaba: 2.2.3.RELEASE

**Comment by yuhuangbin** [comment]
refer #1723

**Comment by kakasiji** [comment]
同样遇到这个问题 1723和1776应该是一样的原因, 请问有解决方案吗

**Comment by yuhuangbin** [comment]
The error occurs when the service is stopped and does not affect normal use.

**Comment by f00lish** [comment]
> The error occurs when the service is stopped and does not affect normal use.

会影响服务关闭从nacos下线的时间吧，这段时间还是在线状态，其他服务调用会报错 有办法么

**Comment by JMMMM** [comment]
同遇到这个问题

**Comment by FalconIA** [comment]
Fixed.
It reappears in `2021.1`, and fixed by #2028 (which is not released yet).
